android tests failing because appium's apidemos is crashing

Is anyone else experiencing this? @bootstraponline @0x1mason @imurchie 

Basically, certain appium tests try to start our apidemos with a certain activity (for example, `io.appium.android.apis/.app.StatusBarNotifications`). The app attempts to launch this activity, then it immediately crashes. Below is the logcat exception.

A number of tests are failing this way---the notifications tests and the touch action tests, most notably.

Adding to 1.3.5 since it seems like a pretty serious bug. I get it after a pull from master and a clean reset --dev --android so it seems legit.

```
E/AndroidRuntime( 3241): FATAL EXCEPTION: main
E/AndroidRuntime( 3241): Process: io.appium.android.apis, PID: 3241
E/AndroidRuntime( 3241): java.lang.RuntimeException: Unable to start activity ComponentInfo{io.appium.android.apis/io.appium.android.apis.app.StatusBarNotifications}: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.Button.setOnClickListener(android.view.View$OnClickListener)' on a null object reference
E/AndroidRuntime( 3241):        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2298)
E/AndroidRuntime( 3241):        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2360)
E/AndroidRuntime( 3241):        at android.app.ActivityThread.access$800(ActivityThread.java:144)
E/AndroidRuntime( 3241):        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278)
E/AndroidRuntime( 3241):        at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime( 3241):        at android.os.Looper.loop(Looper.java:135)
E/AndroidRuntime( 3241):        at android.app.ActivityThread.main(ActivityThread.java:5221)
E/AndroidRuntime( 3241):        at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 3241):        at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime( 3241):        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
E/AndroidRuntime( 3241):        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
E/AndroidRuntime( 3241): Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.Button.setOnClickListener(android.view.View$OnClickListener)' on a null object reference
E/AndroidRuntime( 3241):        at io.appium.android.apis.app.StatusBarNotifications.onCreate(StatusBarNotifications.java:54)
E/AndroidRuntime( 3241):        at android.app.Activity.performCreate(Activity.java:5933)
E/AndroidRuntime( 3241):        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105)
E/AndroidRuntime( 3241):        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2251)
E/AndroidRuntime( 3241):        ... 10 more
W/ActivityManager( 1307):   Force finishing activity io.appium.android.apis/.app.StatusBarNotifications
```

Does this include the cache change I made?
On Jan 8, 2015 6:23 PM, "Jonathan Lipps" <notifications@github.com> wrote:

> Is anyone else experiencing this? @bootstraponline
> <https://github.com/bootstraponline> @0x1mason
> <https://github.com/0x1mason> @imurchie <https://github.com/imurchie>
>
> Basically, certain appium tests try to start our apidemos with a certain
> activity (for example, io.appium.android.apis/.app.StatusBarNotifications).
> The app attempts to launch this activity, then it immediately crashes.
> Below is the logcat exception.
>
> A number of tests are failing this way---the notifications tests and the
> touch action tests, most notably.
>
> Adding to 1.3.5 since it seems like a pretty serious bug. I get it after a
> pull from master and a clean reset --dev --android so it seems legit.
>
> E/AndroidRuntime( 3241): FATAL EXCEPTION: main
> E/AndroidRuntime( 3241): Process: io.appium.android.apis, PID: 3241
> E/AndroidRuntime( 3241): java.lang.RuntimeException: Unable to start activity ComponentInfo{io.appium.android.apis/io.appium.android.apis.app.StatusBarNotifications}: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.Button.setOnClickListener(android.view.View$OnClickListener)' on a null object reference
> E/AndroidRuntime( 3241):        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2298)
> E/AndroidRuntime( 3241):        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2360)
> E/AndroidRuntime( 3241):        at android.app.ActivityThread.access$800(ActivityThread.java:144)
> E/AndroidRuntime( 3241):        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1278)
> E/AndroidRuntime( 3241):        at android.os.Handler.dispatchMessage(Handler.java:102)
> E/AndroidRuntime( 3241):        at android.os.Looper.loop(Looper.java:135)
> E/AndroidRuntime( 3241):        at android.app.ActivityThread.main(ActivityThread.java:5221)
> E/AndroidRuntime( 3241):        at java.lang.reflect.Method.invoke(Native Method)
> E/AndroidRuntime( 3241):        at java.lang.reflect.Method.invoke(Method.java:372)
> E/AndroidRuntime( 3241):        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
> E/AndroidRuntime( 3241):        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
> E/AndroidRuntime( 3241): Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.Button.setOnClickListener(android.view.View$OnClickListener)' on a null object reference
> E/AndroidRuntime( 3241):        at io.appium.android.apis.app.StatusBarNotifications.onCreate(StatusBarNotifications.java:54)
> E/AndroidRuntime( 3241):        at android.app.Activity.performCreate(Activity.java:5933)
> E/AndroidRuntime( 3241):        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105)
> E/AndroidRuntime( 3241):        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2251)
> E/AndroidRuntime( 3241):        ... 10 more
> W/ActivityManager( 1307):   Force finishing activity io.appium.android.apis/.app.StatusBarNotifications
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/appium/appium/issues/4368>.
>

--- a/grunt-helpers.js
+++ b/grunt-helpers.js
@@ -598,7 +598,10 @@ var getSelendroidVersion = function (cb) {
 
 module.exports.buildAndroidApp = function (grunt, appName, cb) {
   var appPath = path.resolve(__dirname, "sample-code", "apps", appName);
-  buildAndroidProj(grunt, appPath, "debug", cb);
+  buildAndroidProj(grunt, appPath, "clean", function (err) {
+    if (err) return cb(err);
+    buildAndroidProj(grunt, appPath, "debug", cb);
+  });
 };
 
 module.exports.buildSelendroidAndroidApp = function (grunt, appName, cb) {

